dev/test/collection.ts

```diff
@@ -169,4 +175,51 @@ describe('Collection interface', () => {
     expect(coll1.isEqual(coll1Equals)).to.be.ok;
     expect(coll1.isEqual(coll2)).to.not.be.ok;
   });
+
+  it('for CollectionReference.withConverter()', async () => {
+    const doc = document('documentId', 'author', 'author', 'title', 'post');
+    const overrides: ApiOverride = {
+      commit: request => {
+        const expectedRequest = set({
+          document: doc,
+        });
+        requestEquals(request, expectedRequest);
+
+        return response(writeResult(1));
+      },
+      batchGetDocuments: () => {
+        const stream = through2.obj();
+        setImmediate(() => {
+          stream.push({found: doc, readTime: {seconds: 5, nanos: 6}});
+          stream.push(null);
+        });
+
+        return stream;
+      },
+    };
+
+    return createInstance(overrides).then(async firestore => {
+      const docRef = firestore
+        .collection('collectionId')
+        .withConverter(postConverter)
+        .doc('documentId');
+
+      await docRef.set(new Post('post', 'author'));
+      const postData = await docRef.get();
```

Comment:
Should this use CollectionReference.get() since this test is meant to exercise the CollectionReference API? (You can also drop it since you are already testing this in query.ts)

Reply:
I don't think `CollectionReference.get()` exists? 😮 

This test makes susre that calling `CollectionReference.doc()` returns a DocumentReference that has the converter.

Reply:
There is a `Query.get()` that is available via the `CollectionReference` API. 

If you want to test `CollectionReference.doc() ` can you update the name of the test?